airflow/models/baseoperator.py

```diff
@@ -1184,7 +1197,29 @@ def render_template_fields(
         """
         if not jinja_env:
             jinja_env = self.get_template_env()
-        self._do_render_template_fields(self, self.template_fields, context, jinja_env, set())
+
+        if self.template_all_fields:
+            templatable_fields = (
+                set(self.__dict__.keys())
+                - set(BaseOperator.__dict__.keys())
+                - set(partial(operator_class=self.__class__, task_id=self.task_id).kwargs.keys())
+            )
```

Comment:
Was wondering if anyone has any comment on how I did this. Goal was to get attributes that belong only to child classes of `BaseOperator`, would like to know if this approach is satisfactory, or if there is a better way to accomplish it.

Reply:
Well I guess better create some restricted frozen set and keep all parameters which never rendered (I think all of BaseOperator) we also could use this set as reference for end users. It is quite common question "Why I can't render `pool`, `executor_config`, `retries` and etc"

Reply:
And I think we need to exclude all implicit attributes which started with `_`.

Reply:
And idea for another PR: Deprecated all explicit `templated_fields` which started with `_`.

🙄 
https://github.com/apache/airflow/blob/98221da0d96b102b009d422870faf7c5d3d931f4/airflow/providers/apache/spark/operators/spark_submit.py#L75-L90

Reply:
And just wondering is it work with properties?

```python
class OperatorWithProperties(BaseOperator):

    @property
    def hook(self):
        return "foo"

    @cached_property
    def cached_hook(self):
        return "bar"

    def execute(self, context):
        pass
```

I think we should skip all properties